### PR TITLE
refactor(odd): TooManyPinsModal is now SmallModalChildren and uses Modal base

### PR DESCRIPTION
**Pull request by jerader**
closes RAUT-384

# Overview

`TooManyPinsModal` is deleted and replacted with `SmallModalChildren` and now uses `Modal` in ODD as as base modal instead of the `ModalShell`.

# Test Plan

- should match this design for simple small modal: https://www.figma.com/file/OIdG64Q5cgvEw82ish5Eee/Design-System-(Flex)?node-id=145-4583&t=mJyaDEkozFDoKcm1-0 . Looks at storybook and usages in the odd.
- 
Screen Shot 2023-04-10 at 5 53 28 PM

# Changelog

- delete `TooManyPins` modal and replace with `SmallModalChildren`, replacing test and story
- change the 2 usages of `TooManyPins` to instead use `SmallModalChildren`

# Review requests

- see test plan

# Risk assessment

low

### CODE REVIEW

**Review by ewagoner — APPROVED** [review]

File: app/src/molecules/Modal/OnDeviceDisplay/SmallModalChildren.tsx [hunk 1]
```diff
@@ -10,24 +9,25 @@ import {
...
-    >
+    <Modal onOutsideClick={handleCloseMaxPinsAlert} modalSize="small">
       <Flex
         flexDirection={DIRECTION_COLUMN}
         gridGap={SPACING.spacing3}
-        padding={SPACING.spacingXXL}
+        width="100%"
       >
         <StyledText
           color={COLORS.darkBlackEnabled}
```
Comment (koji) [inline app/src/molecules/Modal/OnDeviceDisplay/SmallModalChildren.tsx:33]: The default color is `darkBlackEnabled` so we can remove this if you want.

```suggestion
          
```

File: app/src/pages/OnDeviceDisplay/ProtocolDashboard/LongPressModal.tsx [hunk 2]
```diff
@@ -102,7 +102,10 @@ export function LongPressModal(props: {
   return (
     <>
       {showMaxPinsAlert ? (
-        <TooManyPinsModal
+        <SmallModalChildren
+          header={t('too_many_pins_header')}
+          subText={t('too_many_pins_body')}
+          buttonText={t('shared:close')}
```
Comment (koji) [inline app/src/pages/OnDeviceDisplay/ProtocolDashboard/LongPressModal.tsx:108]: shared close is `close` so we need to make the first letter capitalize.
```suggestion
          buttonText={i18n.format(t('close'), 'sentenceCase')}
```

File: app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx [hunk 3]
```diff
@@ -294,7 +294,10 @@ export function ProtocolDetails(): JSX.Element | null {
   return (
     <Flex flexDirection={DIRECTION_COLUMN} padding={SPACING.spacing6}>
       {showMaxPinsAlert && (
-        <TooManyPinsModal
+        <SmallModalChildren
+          header={t('too_many_pins_header')}
+          subText={t('too_many_pins_body')}
+          buttonText={t('shared:close')}
```
Comment (koji) [inline app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx:300]: same as the above

```suggestion
          buttonText={i18n.format(t('close'), 'sentenceCase')}
```

**Review by koji — APPROVED** [review]
Looks good to me.
The comment on StyledText is completely optional.

**Pull request merged**